src/pal/pal_sk_posix.c

```diff
@@ -1195,7 +1195,7 @@ int32_t pal_socket_getsockopt(
     int32_t opt_lvl, opt_name;
     socklen_t opt_len;
     int32_t opt_val;
-    u_long avail;
+    prx_size_t avail;
```

Comment:
Yeah, good one, what I could find is that IOCtl takes a pointer to an int, so prx_size_t should work.  However, I might decide to typedef prx_size_t to size_t or even replace it. so, I would say u_long might better change to int, then assinging it to *value should be fine.